src/foam/u2/view/EditColumnsView.js

```diff
@@ -4,84 +4,124 @@
  * http://www.apache.org/licenses/LICENSE-2.0
  */
 
+foam.ENUM({
+  package: 'foam.u2.view',
+  name: 'ColumnVisibility',
+  values: [
+    {
+      name: 'DEFAULT',
+      label: 'Default'
+    },
+    {
+      name: 'ALWAYS_SHOW',
+      label: 'Always Show'
+    },
+    {
+      name: 'ALWAYS_HIDE',
+      label: 'Always Hide'
+    },
+  ]
+});
+
 foam.CLASS({
   package: 'foam.u2.view',
-  name: 'EditColumnsView',
-  extends: 'foam.u2.Element',
+  name: 'ColumnConfig',
+  sections: [{ name: '_defaultSection' }],
+  properties: [
+    {
+      class: 'String',
+      name: 'name',
+      hidden: true
+    },
+    {
+      class: 'String',
+      name: 'label',
+      label: '',
+      visibility: 'RO',
+      gridColumns: 6
+    },
+    {
+      class: 'Enum',
+      of: 'foam.u2.view.ColumnVisibility',
+      name: 'visibility',
+      label: '',
+      gridColumns: 6
+    }
+  ]
+});
 
+foam.CLASS({
+  package: 'foam.u2.view',
+  name: 'EditColumnsView',
```

Comment:
This isn't a view anymore, right? Might be better to rename it.

Reply:
Good point. As of https://github.com/foam-framework/foam2/pull/2455/files it is toE-able which makes it more like a view so I think it's ok to keep calling it that.